bluemira/radiation_transport/radiation_profile.py

```diff
@@ -0,0 +1,2151 @@
+            Values > 1 for the scrape-off layer
+        rad_power: np.array
+            radiated power at each mid-plane location corresponding to rho [Mw/m^3]
+        imp_name: [strings]
+            impurity neames
+        """
+        if ax is None:
+            fig, ax = plt.subplots()
+            plt.xlabel(r"$\rho$")
+            plt.ylabel(r"$[MW.m^{-3}]$")
+
+        if len(rad_power) == 1:
+            ax.plot(rho, rad_power)
+        else:
+            [
+                ax.plot(rho, rad_part, label=name)
+                for rad_part, name in zip(rad_power, imp_name)
+            ]
+            rad_tot = np.sum(np.array(rad_power, dtype=object), axis=0).tolist()
+            plt.title("Core radiated power density")
+            ax.plot(rho, rad_tot, label="total radiation")
+            ax.legend(loc="best", borderaxespad=0, fontsize=12)
+
+        return ax
+
+
+class CoreRadiation(Radiation):
+    """
+    Specific class to calculate the core radiation source.
+    Temperature and density are assumed to be constant along a
+    single flux tube.
+    """
+
+    # fmt: off
+    default_params = Radiation.default_params + [
+        ["n_e_0", "Electron density on axis", 1.81e+20, "1/m^3", None, "Input"],
+        ["T_e_0", "Electron temperature on axis", 2.196e+01, "keV", None, "Input"],
+        ["rho_ped_n", "Density pedestal r/a location", 9.4e-01, "dimensionless", None, "Input"],
+        ["rho_ped_t", "Temperature pedestal r/a location", 9.76e-01 , "dimensionless", None, "Input"],
+        ["n_e_ped", "Electron density pedestal height", 1.086e+20, "1/m^3", None, "Input"],
+        ["T_e_ped", "Electron temperature pedestal height", 3.74, "keV", None, "Input"],
+        ["alpha_n", "Density profile factor", 1.15, "dimensionless", None, "Input"],
+        ["alpha_t", "Temperature profile index", 1.905, "dimensionless", None, "Input"],
+        ["t_beta", "Temperature profile index beta", 2, "dimensionless", None, "Input"],
+    ]
+    # fmt: on
+
+    def __init__(
+        self,
+        eq: Equilibrium,
+        flux_surf_solver: TempFsSolver,
+        params: ParameterFrame,
+        psi_n,
+        ne_mp,
+        te_mp,
+        impurity_content,
+        impurity_data,
+    ):
+        super().__init__(eq, flux_surf_solver, params)
+
+        self.H_content = impurity_content["H"]
+        self.impurities_content = [
+            frac for key, frac in impurity_content.items() if key != "Ar"
+        ]
+        self.imp_data_t_ref = [
+            data["T_ref"] for key, data in impurity_data.items() if key != "Ar"
+        ]
+        self.imp_data_l_ref = [
+            data["L_ref"] for key, data in impurity_data.items() if key != "Ar"
+        ]
+        self.impurity_symbols = impurity_content.keys()
+
+        # Midplane profiles
+        self.psi_n = psi_n
+        self.ne_mp = ne_mp
+        self.te_mp = te_mp
+
+    def build_mp_radiation_profile(self):
```

Comment:
I thought all of these profiles were meant to be coming in as inputs now? Something like this object or its children in equilibria:
https://github.com/Fusion-Power-Plant-Framework/bluemira/blob/develop/bluemira/equilibria/profiles.py#L363